minePlace.js

```diff
@@ -0,0 +1,114 @@
+class MinePlace {
+  minesAmount;
+
+  randomMinesTemp = [];
+
+  mines = [];
+
+  minesForStopwatch = [];
+
+  minesLeftCounter = [];
+
+  flagCounter = 0;
+
+  constructor() {
+    this.minePlace();
+  }
+
+  minePlace() {
+    const random = () => {
+      const boardSize = [];
+      const randomMinesTemp = [];
+      for (let i = 0; i < board.size; i++) {
+        boardSize.push(i);
+      }
+      for (boardSize, this.i = boardSize.length; this.i--;) {
+        const randomNum = boardSize.splice(Math.floor(Math.random() * (this.i + 1)), 1)[0];
+        randomMinesTemp.push(randomNum); // making a new array with digits from previous array
+        // but in random order
+      }
+      return randomMinesTemp;
+    };
+    this.randomMinesTemp = random();
+    this.minesAmount = (board.size) / 8;
+    this.minesAmount = Math.floor(this.minesAmount);
+    this.minesLeftCounter = this.minesAmount;
+    for (let i = 0; i < this.minesAmount; i++) {
+      // picking first digits to place the mines. Amount of mines defined by the board size
+      const rand = this.randomMinesTemp[this.randomMinesTemp.length - 1 - i];
+      this.minesForStopwatch.push(rand);
+      // will be used later on to stop the timer if button with mine is pressed
+      this.mines.push(rand);
+    }
+    for (let i = 0; i < this.mines.length; i++) {
+      // placing mines and changing the field's property "mine"
+      board.fieldsList[this.mines[i]].mine = true;
+    }
+    for (let i = 0; i < board.size; i++) {
```

Comment:
the number of the mines should be selected base on the difficulty level. And you should place selected number of mines on the board. So in that case there is no need to count mines.